src/utils/source-queue.js

```diff
@@ -32,6 +32,16 @@ export default {
     queuedSources.push(source);
     queue();
   },
+  queueSources: sources => {
+    queuedSources = queuedSources.concat(
+      sources.map(source => {
+        source.actor = source.id;
```

Comment:
Thanks for fixing the tests!

Why is this required here tho? Shouldn't the callers set source.actor if it's required?

Reply:
There are few possibilities where we can add actor as property, or use `id` everywhere(it's what we should do in future for CDP compatibility). I just wanted to see if this fixes things.

We can either say `source.actor || source.id` [here](https://github.com/devtools-html/debugger.html/blob/master/src/client/firefox/create.js#L48)
and fix flow, this would be what I'd do but flow is upset with me.

Reply:
The type `Source` defines an [id property](https://github.com/devtools-html/debugger.html/blob/fcff67bcf43a118ba7c967a0445df9006b75176a/src/types.js#L294) while `SourcePayload` defines an [actor property](https://github.com/devtools-html/debugger.html/blob/5bc3da76eab9ac5d160c88e1dc5aaeb63b56f20a/src/client/firefox/types.js#L86). Is SourcePayload supposed to be a serialized Source? If so SourcePayload.actor should probably become SourcePayload.id?

Reply:
So, SourcePayload is for what we get from the server, that is, source id as `actor` property.
But on the client side, we don't use the actor property instead we use `id`.
Ideally, we'd want the server to send us `id` and not `actor`
But for now, we'd have to make a patch somewhere.

Reply:
So we call this `createSource` [function](https://github.com/devtools-html/debugger.html/blob/master/src/client/firefox/create.js#L43) when we get a new source from the server, which has the actor property.

But we also use this function to create a source from original urls(which are generated on client side) and has `id` instead of `actor`. So those original sources gets generated without an id, which causes further issues.

I have not looked into it in much detail as to find a nice solution. But I am confident that you'll find some nice answer.

Reply:
Thanks that really helped me understand the problem!

The problem is that now the same `sourceQueue` is used to create sources coming from the firefox server (received as instances of type `SourcePayload`), and from the client (instances of type `Source`). That's problematic since the types don't have matching fields, and the client is supposed to abstract over server implementations anyway. 

One possible solution is to refactor `source-queue.js` so that it's not a module-singleton and use two separate queues, one for sources coming from the server and another for sources coming from the client. There's other simpler solutions similar to your fix, but applied to `src/client/firefox`. @jasonLaster opinions?

Reply:
Good question. 

I think I would change the newSource function so it calls createSource before enquing the source. 

This should be simpler too